File: docs/breez-sdk/snippets/swift/BreezSdkSnippets/Sources/PreparePaymentLink.swift
import BreezSdkSpark
import Foundation

func preparePaymentLinkViaCashapp(sdk: BreezSdk) async throws {
    // ANCHOR: prepare-payment-link-cashapp
    // Parse the recipient's external-chain address (EVM/Solana/Tron).
    let parsed = try await sdk.parse(input: "<recipient address>")
    guard case let .crossChainAddress(v1: addressDetails) = parsed else {
        throw NSError(domain: "PreparePaymentLink", code: 1)
    }

    // List the stablecoin destinations you can send to and pick one, e.g. USDC on Base.
    // Restrict to routes fundable over Lightning.
    let routes = try await sdk.getCrossChainRoutes(
        filter: .paymentLink(addressDetails: addressDetails))
    guard let route = routes.first(where: { $0.asset == "USDC" && $0.chain == "base" }) else {
        throw NSError(domain: "PreparePaymentLink", code: 2)
    }

    // Send $10 of USDC, funded by Cash App over Lightning. The amount is in the
    // route asset's base units (USDC, 6 decimals), so 10_000_000 = 10 USDC, about $10.
    let response = try await sdk.preparePaymentLink(
        request: PreparePaymentLinkRequest(
            address: addressDetails.address,
            route: route,
            amount: 10_000_000,
            feePolicy: nil,
            maxSlippageBps: nil
        ))
    print("Open this URL in Cash App: \(response.url)")
    print("Recipient receives ~\(response.estimatedOut) \(response.asset)")
    // ANCHOR_END: prepare-payment-link-cashapp
}

File: docs/breez-sdk/snippets/flutter/lib/prepare_payment_link.dart
import 'package:breez_sdk_spark_flutter/breez_sdk_spark.dart';

Future<void> preparePaymentLinkViaCashapp(BreezSdk sdk) async {
  // ANCHOR: prepare-payment-link-cashapp
  // Parse the recipient's external-chain address (EVM/Solana/Tron).
  String input = "<recipient address>";
  InputType parsed = await sdk.parse(input: input);
  if (parsed is! InputType_CrossChainAddress) {
    throw Exception("Not a cross-chain address");
  }
  CrossChainAddressDetails addressDetails = parsed.field0;

  // List the stablecoin destinations you can send to and pick one, e.g. USDC
  // on Base. These routes are funded by an external rail, not the Spark wallet.
  // Restrict to Lightning-fundable routes to match the Cash App flow below.
  List<CrossChainRoutePair> routes = await sdk.getCrossChainRoutes(
    filter: CrossChainRouteFilter_PaymentLink(
      addressDetails: addressDetails,
    ),
  );
  CrossChainRoutePair route =
      routes.firstWhere((r) => r.asset == "USDC" && r.chain == "base");

  // Send $10 of USDC, funded by Cash App over Lightning. The amount is in the
  // route asset's base units (USDC, 6 decimals), so 10000000 = 10 USDC,
  // about $10.
  final response = await sdk.preparePaymentLink(
    request: PreparePaymentLinkRequest(
      address: addressDetails.address,
      route: route,
      amount: BigInt.from(10000000),
      feePolicy: null,
      maxSlippageBps: null,
    ),
  );

  // Open this Cash App URL to pay; the recipient then receives the stablecoin.
  print("Open this URL in Cash App: ${response.url}");
  print("Recipient receives ~${response.estimatedOut} ${response.asset}");
  // ANCHOR_END: prepare-payment-link-cashapp
}
